src/App/Fossa/Config/Project/Edit.hs

```diff
@@ -0,0 +1,135 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.Config.Project.Edit (
+  EditConfig (..),
+  EditOpts (..),
+  subcommand,
+  cliParser,
+  mergeOpts,
+) where
+
+import App.Fossa.Config.Common (configHelp, endpointHelp, fossaApiKeyCmdText, fossaApiKeyHelp, parsePolicyOptions, titleHelp, validateApiKeyGeneric)
+import App.Fossa.Config.ConfigFile (ConfigFile (..), ConfigProject (..))
+import App.Fossa.Config.EnvironmentVars (EnvVars (envApiKey))
+import App.OptionExtensions (uriOption)
+import App.Types (Policy (..), ProjectMetadataRevision (..))
+import Control.Effect.Diagnostics (Diagnostics, Has, fatalText)
+import Data.Aeson (ToJSON, defaultOptions, genericToEncoding, toEncoding)
+import Data.Text (Text)
+import Data.Text qualified
+import Fossa.API.Types (ApiOpts (..), defaultApiPollDelay)
+import GHC.Generics (Generic)
+import Options.Applicative (
+  CommandFields,
+  InfoMod,
+  Mod,
+  Parser,
+  command,
+  helpDoc,
+  info,
+  long,
+  many,
+  metavar,
+  optional,
+  short,
+  strOption,
+  switch,
+  (<|>),
+ )
+import Options.Applicative.Builder (progDescDoc)
+import Style (applyFossaStyle, formatStringToDoc, stringToHelpDoc)
+import Text.URI (URI, mkURI)
+
+projectEditInfo :: InfoMod a
+projectEditInfo = progDescDoc $ formatStringToDoc "Edit a FOSSA project"
+
+subcommand :: (EditOpts -> a) -> Mod CommandFields a
+subcommand f = command "edit" $ info (f <$> cliParser) projectEditInfo
+
+data EditConfig = EditConfig
+  { apiOpts :: ApiOpts
+  , projectLocator :: Text
+  , projectMetadataRevision :: ProjectMetadataRevision
+  }
+  deriving (Eq, Ord, Show, Generic)
+
+instance ToJSON EditConfig where
+  toEncoding = genericToEncoding defaultOptions
+
+data EditOpts = EditOpts
+  { debug :: Bool
+  , baseUrlOpts :: Maybe URI
+  , apiKeyOpts :: Maybe Text
+  , configOpts :: Maybe FilePath
+  , projectLocatorOpts :: Maybe Text
+  , projectMetadataRevisionOpts :: ProjectMetadataRevision
```

Comment:
[nit] I think it's understood from the type-name that these are options, so no need for the `Opts` suffix. It's particularly confusing for the fields that are really just single values IMO.

`configOpts` in particular should probably be something more like `configFile`.